DS-139: convert everything to use new CSS spacing vars

mikemai2awesome opened the pull request:
## Jira

https://pegadigitalit.atlassian.net/browse/DS-139

## Summary

Removed all usage of `bolt-margin*`, `bolt-padding*`, `bolt-spacing`, and `bolt-v-spacing` and converted to use `var(--bolt-spacing-x-*)` and `var(--bolt-spacing-y-*)` instead.

## Details

1. All components are converted.
1. All global and core files converted.
1. All docs-site files converted.
1. Removed old json maps.

## How to test

Run locally and do visual test on every single page in PL.

## Release notes

### Deprecated SCSS mixins:

1. `bolt-margin`
1. `bolt-padding`
1. `bolt-margin-*`
1. `bolt-padding-*`

### Deprecated SCSS functions

1. `bolt-spacing`
1. `bolt-v-spacing`

### New spacing vars usage

Note: `x` represents horizontal, `y` represents vertical.

#### All directions

```scss
margin: var(--bolt-spacing-y-medium) var(--bolt-spacing-x-medium);
padding: var(--bolt-spacing-y-medium) var(--bolt-spacing-x-medium);
```

#### One direction

```scss
margin-top: var(--bolt-spacing-y-medium);
margin-right: var(--bolt-spacing-x-medium);
padding-top: var(--bolt-spacing-y-medium);
padding-right: var(--bolt-spacing-x-medium);
top: var(--bolt-spacing-y-medium);
right: var(--bolt-spacing-x-medium);
```

#### Math
```scss
margin-top: calc(var(--bolt-spacing-y-medium) - 6px);
margin-right: calc(var(--bolt-spacing-x-medium) * 666);
```

colbytcook commented on docs-site/src/components/off-canvas-nav/off-canvas-nav.scss:
```diff
@@ -34,7 +34,7 @@
     &--active-path {
       // color: #545DA6;
       // font-weight: 600;
```
should we delete the commented out code here?

mikemai2awesome replied:
yes, it's not doing anything and we have no bugs while they are commented out currently.

colbytcook commented on docs-site/src/components/off-canvas-nav/off-canvas-nav.scss:
```diff
@@ -82,7 +82,7 @@
       // margin-top: 0.25rem;
       // margin-bottom: 0.25rem;
       // border-left: 2px solid transparent;
```
should we delete the commented out code here?

mikemai2awesome replied:
yes, it's not doing anything and we have no bugs while they are commented out currently.

colbytcook commented on docs-site/src/components/off-canvas-nav/off-canvas-nav.scss:
```diff
@@ -82,7 +82,7 @@
       // border-left: 2px solid transparent;
-      @include bolt-margin-left(small);
+      margin-left: var(--bolt-spacing-x-small);
       // padding-left: 2rem;
```
should we delete the commented out code here?

mikemai2awesome replied:
yes, it's not doing anything and we have no bugs while they are commented out currently.

colbytcook commented on docs-site/src/pages/pattern-lab/_patterns/04-pages/99999-bolt-dev-sandbox/styles/_global-search.scss:
```diff
@@ -99,7 +100,7 @@ $bolt-search-field-icon-only-button-width: bolt-spacing(medium) + 1rem;
   order: 2;
   transform: translate3d(
-    ($bolt-search-field-icon-only-button-width * -1),
+    calc(#{$bolt-search-field-icon-only-button-width} * -1),
```
should this have the "var(--" instead of the '#{'?

colbytcook commented on packages/experimental/micro-journeys/src/nav-shared.scss:
```diff
@@ -37,7 +37,7 @@ $nav-circle-border-size: 2px;
         2
     );
-    margin-right: bolt-spacing(small);
+    margin-right: bvar(--bolt-spacing-x-small);
```
typo

mikemai2awesome replied:
fixed

mikemai2awesome commented:
@danielamorse I addressed all your comments except for dropdown, that's an intentional fix. I don't think that's being used anywhere, so the very minor difference shouldn't matter.
